extensions/amp-list/0.1/amp-list.js

```diff
@@ -1109,6 +1163,46 @@ export class AmpList extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Measure and lock height before performing given mutate fn.
+   * Applicable for amp-list initialized with layout=container.
+   * @private
+   * @param {!Function} mutate
+   * @return {!Promise}
+   */
+  lockHeightAndMutate_(mutate) {
+    devAssert(
```

Comment:
Note that devAssert is stripped in prod builds so this won't function as an escape hatch if that's your intent.

Reply:
Thanks for this note. I thought about adding:
```js
    if (!this.enableManagedResizing_ || this.loadMoreEnabled_) {
      return this.mutateElement(mutate);
    }
```

as a replacement or addendum to the assertion to act as an actual guard. However, I'm not sure either this `return` or the ineffectual alternative `return Promise.resolve()` make sense in this case. Really it should not be called if either of the included conditions is `true`, which is the case where the method is used now and would more be a guard against future misuse.

Alternatively we could modify the `devAssert` with a `userAssert`, right? Or literally `throw Error`?

Reply:
Yea I think it's fine to have a defense in depth here if we're a bit paranoid. :) 

I'd do the conditional you suggest plus `dev().error` and an early return.